### ISSUE TITLE
syslog plugin is broken on node 6

### ISSUE THREAD

**Issue by msimerson** [haraka/Haraka#1546]
### Haraka version

2.8.8

### Expected behavior

Haraka to start up normally.

````
....
[NOTICE] [-] [core] Daemonizing...
````

### Observed behavior

````
loglevel: LOGINFO
Starting up Haraka version 2.8.8
[INFO] [-] [core] Loading plugins
[INFO] [-] [core] Loading plugin: log.syslog
[INFO] [-] [core] Loading plugin: auth/auth_vpopmaild
[INFO] [-] [core] Loading plugin: rcpt_to.qmail_deliverable
[INFO] [-] [core] Loading plugin: queue/smtp_forward
Jul 20 22:20:56 stage haraka[9577]: [NOTICE] [-] [core] Daemonizing...
Jul 20 22:20:56 stage haraka[9578]: [NOTICE] [-] [core] Listening on 172.16.15.254:25
Jul 20 22:20:56 stage haraka[9578]: [NOTICE] [-] [core] Listening on 172.16.15.254:465
Jul 20 22:20:56 stage haraka[9578]: [NOTICE] [-] [core] Listening on 172.16.15.254:587
Jul 20 22:20:56 stage haraka[9578]: [INFO] [-] [core] [outbound] Loading outbound queue from /usr/local/haraka/queue
Jul 20 22:20:56 stage haraka[9578]: [INFO] [-] [core] express loaded at Server.http.express
Jul 20 22:20:56 stage haraka[9578]: [INFO] [-] [core] express app is at Server.http.app
Jul 20 22:20:56 stage haraka[9578]: [NOTICE] [-] [core] Listening on 172.16.15.254:80
Jul 20 22:20:56 stage haraka[9578]: [INFO] [-] [core] init_http_respond
Jul 20 22:20:56 stage haraka[9578]: [INFO] [-] [core] Server.http.wss loaded
Jul 20 22:20:56 stage haraka[9578]: [INFO] [-] [core] init_wss_respond
Jul 20 22:20:56 stage haraka[9578]: [INFO] [-] [core] using html docroot: /usr/local/haraka/html
Jul 20 22:20:56 stage haraka[9578]: [CRIT] [-] [core] TypeError: undefined is not a function
Jul 20 22:20:56 stage haraka[9578]: [CRIT] [-] [core] TypeError: undefined is not a function
Jul 20 22:20:56 stage haraka[9578]: [CRIT] [-] [core] TypeError: undefined is not a function
Jul 20 22:20:56 stage haraka[9578]: [CRIT] [-] [core] TypeError: undefined is not a function
Jul 20 22:20:56 stage haraka[9578]: [NOTICE] [-] [core] Shutting down
````

### Steps to reproduce

1. install Haraka
2. enable log.syslog plugin
3. start Haraka

**Comment by msimerson** [comment]
On the same FreeBSD host, I have Haraka 2.8.6 installed with no such issues. I haven't been able to determine why.

**Comment by msimerson** [comment]
another clue: reverting to node 4.4.7 works.

**Comment by msimerson** [comment]
see strongloop/modern-syslog#16

**Comment by Infern1** [comment]
Seeing the same thing
Node 6.2.2

**Comment by zombified** [comment]
FWIW I ran into this on `0.10.46-1nodesource1~trusty1` package from the [nodesource deb repo](https://deb.nodesource.com/node_0.10/).

Ended up installing `5.12.0-1nodesource1~trusty1` from their node_5.x repo, and it stopped giving me the error.

I also noticed that restarting the Haraka process several times in a row would often (but not always) yield a different number of duplicate `TypeError: undefined is not a function` messages.

When I altered the `always_ok` value in my `config/log.syslog.ini` file to `true` the haraka process would start and then just quit without any apparent error message.
